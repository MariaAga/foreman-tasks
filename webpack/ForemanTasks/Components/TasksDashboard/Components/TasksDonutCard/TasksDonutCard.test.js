import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import TasksDonutCard from './TasksDonutCard';

const createRequiredProps = () => ({ last: 3, older: 5 });

const fixtures = {
  'render with minimal props': { ...createRequiredProps() },
  'render with props': {
    ...createRequiredProps(),
    title: 'some title',
    className: 'some-classname',
    focusedOn: 'normal',
    onTotalClick: jest.fn(),
  },
};

TasksDonutCard.focusedOnOptions.forEach(mode => {
  fixtures[`render with focused-on ${mode}`] = {
    ...createRequiredProps(),
    focusedOn: mode,
  };
});

describe('TasksDonutCard', () =>
  testComponentSnapshotsWithFixtures(TasksDonutCard, fixtures));
