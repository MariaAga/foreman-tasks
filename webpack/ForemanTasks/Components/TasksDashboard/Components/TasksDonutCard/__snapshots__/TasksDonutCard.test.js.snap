// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksDonutCard render with focused-on last 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card selected-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
  />
  <CardBody
    className=""
  >
    <TasksDonutChart
      className=""
      colorsPattern={
        Array [
          "#C315C7",
          "#0089C9",
        ]
      }
      focusedOn={
        Object {
          "last": true,
        }
      }
      last={3}
      older={5}
      onLastClick={[Function]}
      onOlderClick={[Function]}
      onTotalClick={[Function]}
      timePeriod="24h"
    />
  </CardBody>
</Card>
`;

exports[`TasksDonutCard render with focused-on none 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card"
  matchHeight={false}
>
  <CardTitle
    className=""
  />
  <CardBody
    className=""
  >
    <TasksDonutChart
      className=""
      colorsPattern={
        Array [
          "#C315C7",
          "#0089C9",
        ]
      }
      focusedOn={
        Object {
          "none": true,
        }
      }
      last={3}
      older={5}
      onLastClick={[Function]}
      onOlderClick={[Function]}
      onTotalClick={[Function]}
      timePeriod="24h"
    />
  </CardBody>
</Card>
`;

exports[`TasksDonutCard render with focused-on normal 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card"
  matchHeight={false}
>
  <CardTitle
    className=""
  />
  <CardBody
    className=""
  >
    <TasksDonutChart
      className=""
      colorsPattern={
        Array [
          "#C315C7",
          "#0089C9",
        ]
      }
      focusedOn={
        Object {
          "normal": true,
        }
      }
      last={3}
      older={5}
      onLastClick={[Function]}
      onOlderClick={[Function]}
      onTotalClick={[Function]}
      timePeriod="24h"
    />
  </CardBody>
</Card>
`;

exports[`TasksDonutCard render with focused-on older 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card selected-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
  />
  <CardBody
    className=""
  >
    <TasksDonutChart
      className=""
      colorsPattern={
        Array [
          "#C315C7",
          "#0089C9",
        ]
      }
      focusedOn={
        Object {
          "older": true,
        }
      }
      last={3}
      older={5}
      onLastClick={[Function]}
      onOlderClick={[Function]}
      onTotalClick={[Function]}
      timePeriod="24h"
    />
  </CardBody>
</Card>
`;

exports[`TasksDonutCard render with focused-on total 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card selected-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
  />
  <CardBody
    className=""
  >
    <TasksDonutChart
      className=""
      colorsPattern={
        Array [
          "#C315C7",
          "#0089C9",
        ]
      }
      focusedOn={
        Object {
          "total": true,
        }
      }
      last={3}
      older={5}
      onLastClick={[Function]}
      onOlderClick={[Function]}
      onTotalClick={[Function]}
      timePeriod="24h"
    />
  </CardBody>
</Card>
`;

exports[`TasksDonutCard render with minimal props 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card"
  matchHeight={false}
>
  <CardTitle
    className=""
  />
  <CardBody
    className=""
  >
    <TasksDonutChart
      className=""
      colorsPattern={
        Array [
          "#C315C7",
          "#0089C9",
        ]
      }
      focusedOn={Object {}}
      last={3}
      older={5}
      onLastClick={[Function]}
      onOlderClick={[Function]}
      onTotalClick={[Function]}
      timePeriod="24h"
    />
  </CardBody>
</Card>
`;

exports[`TasksDonutCard render with props 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card some-classname"
  matchHeight={false}
>
  <CardTitle
    className=""
    onClick={[MockFunction]}
  >
    some title
  </CardTitle>
  <CardBody
    className=""
  >
    <TasksDonutChart
      className=""
      colorsPattern={
        Array [
          "#C315C7",
          "#0089C9",
        ]
      }
      focusedOn={
        Object {
          "normal": true,
        }
      }
      last={3}
      older={5}
      onLastClick={[Function]}
      onOlderClick={[Function]}
      onTotalClick={[MockFunction]}
      timePeriod="24h"
    />
  </CardBody>
</Card>
`;
